apollo-router/src/plugins/response_cache/plugin.rs

```diff
@@ -1620,58 +1620,102 @@ fn get_invalidation_entity_keys_from_schema(
     typename: &str,
     entity_keys: &serde_json_bytes::Map<ByteString, Value>,
 ) -> Result<HashSet<String>, anyhow::Error> {
-    let field_def =
+    let type_def =
         supergraph_schema
             .get_object(typename)
             .ok_or_else(|| FetchError::MalformedRequest {
                 reason: "can't find corresponding type for __typename {typename:?}".to_string(),
             })?;
-    let cache_keys = field_def
+    let implements_cache_keys = type_def
```

Comment:
The validation rule will ensure all implementor types inherit all cache tags from their interfaces. So, we won't need to check cache tags on interfaces.